apps/cowswap-frontend/src/modules/notifications/containers/ConnectTelegram.tsx

```
@@ -14,58 +30,107 @@
   const [isTgSubscribed, setTgSubscribed] = useState<boolean>(false)
+  const [isCmsCallInProgress, setIsCmsCallInProgress] = useState<boolean>(false)
+  const [isLoginInProgress, setIsLoginInProgress] = useState<boolean>(false)
+
   const [tgData, setTgData] = useState<TelegramData | null>(null)
   const telegramWrapperRef = useRef<HTMLDivElement>(null)
   const isDataReceivedRef = useRef(false)
+  const isAuthRequestedRef = useRef(false)
+
+  const authorize = (callback?: () => void) => {
+    getTelegramAuth(TG_BOT_ID, (response) => {
+      if (response && response.user) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)